core/helpers.py

```diff
@@ -11,6 +11,13 @@
 )
 
 
+def instantiate_subclass(base_class, subclass_name):
```

Comment:
Hey,

Out of curiosity, what is this suppose to do?

Thanks!

Reply:
Just return the specified subclass. 

It create all subclasses and return the one matching the argument (key = name of the subclass)

Could be improve actually.

Reply:
OK I think I get it. I **think** it would be better to implement it as a dedicated classmethod of `Observable`, that would be called as `Observable.with_type(typename)`.

Reply:
BTW this function is weird. This should do what you want, without creating an unnecessary `dict`, and without testing all the subclasses.
```Python
    for cls in base_class.__subclasses__():
        if cls.__name__ == subclass_name:
            return cls
```

Reply:
Oh and a better name would be something like: `Observable.class_from_typename(typename)`.

Reply:
I put it in helpers file because it's not specific to Observable class but could work with any other classes

Reply:
It could. You can define it as a subclass of Node if you prefer.